File: scripts/apply_owner_test_mode_settings_migration.mjs
#!/usr/bin/env node
import fs from 'node:fs/promises';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

import dotenv from 'dotenv';
import pg from 'pg';

const { Client } = pg;

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const rootDir = path.resolve(__dirname, '..');

dotenv.config({ path: path.join(rootDir, '.env') });

const migrationPath = path.join(
  rootDir,
  'supabase',
  'migrations',
  '20260327113000_add_owner_test_mode_to_agent_settings.sql',
);

function firstPresent(...values) {
  for (const value of values) {
    const trimmed = String(value || '').trim();
    if (trimmed) return trimmed;
  }
  return '';
}

function buildClientConfig() {
  const connectionString = firstPresent(
    process.env.SUPABASE_DB_URL,
    process.env.DATABASE_URL,
    process.env.POSTGRES_URL,
    process.env.POSTGRES_PRISMA_URL,
    process.env.POSTGRES_URL_NON_POOLING,
  );

  if (connectionString) {
    return {
      connectionString,
      ssl: {
        rejectUnauthorized: false,
      },
    };
  }

  const host = firstPresent(process.env.SUPABASE_DB_HOST);
  const database = firstPresent(process.env.SUPABASE_DB_NAME);
  const user = firstPresent(process.env.SUPABASE_DB_USER);
  const password = firstPresent(process.env.SUPABASE_DB_PASSWORD);
  const port = Number(process.env.SUPABASE_DB_PORT || 0) || 5432;

  if (!host || !database || !user || !password) {
    throw new Error(
      'Missing database connection settings. Define SUPABASE_DB_URL, DATABASE_URL, POSTGRES_URL or SUPABASE_DB_* in .env.',
    );
  }

  return {
    host,
    port,
    database,
    user,
    password,
    ssl: {
      rejectUnauthorized: false,
    },
  };
}

const sql = await fs.readFile(migrationPath, 'utf8');
const client = new Client(buildClientConfig());

try {
  await client.connect();
  await client.query(sql);
  console.log(
    JSON.stringify(
      {
        ok: true,
        migration: path.basename(migrationPath),
      },
      null,
      2,
    ),
  );
} finally {
  await client.end().catch(() => {});
}

File: scripts/apply_owner_test_mode_settings_migration.py
#!/usr/bin/env python3
import os
from pathlib import Path

import pg8000


ROOT = Path(__file__).resolve().parent.parent
MIGRATION_PATH = ROOT / "supabase" / "migrations" / "20260327113000_add_owner_test_mode_to_agent_settings.sql"


def load_env(path: Path) -> None:
    if not path.exists():
        return
    for raw_line in path.read_text().splitlines():
        line = raw_line.strip()
        if not line or line.startswith("#") or "=" not in line:
            continue
        key, value = line.split("=", 1)
        key = key.strip()
        value = value.strip().strip("'").strip('"')
        os.environ.setdefault(key, value)


def first_present(*values: str) -> str:
    for value in values:
        text = str(value or "").strip()
        if text:
            return text
    return ""


def build_connection() -> dict:
    host = first_present(os.environ.get("SUPABASE_DB_HOST"))
    database = first_present(os.environ.get("SUPABASE_DB_NAME"))
    user = first_present(os.environ.get("SUPABASE_DB_USER"))
    password = first_present(os.environ.get("SUPABASE_DB_PASSWORD"))
    port = int(os.environ.get("SUPABASE_DB_PORT") or "5432")

    if not host or not database or not user or not password:
        raise SystemExit(
            "Missing SUPABASE_DB_* settings in .env. This Python migrator expects host/port/db/user/password."
        )

    return {
        "host": host,
        "port": port,
        "database": database,
        "user": user,
        "password": password,
        "ssl_context": True,
    }


def main() -> None:
    load_env(ROOT / ".env")
    sql = MIGRATION_PATH.read_text()
    connection = pg8000.connect(**build_connection())
    try:
        cursor = connection.cursor()
        cursor.execute(sql)
        connection.commit()
        print('{"ok": true, "migration": "20260327113000_add_owner_test_mode_to_agent_settings.sql"}')
    finally:
        connection.close()


if __name__ == "__main__":
    main()
